pkg/networkmanager/firewall-client.js

```diff
@@ -335,6 +335,22 @@ cockpit.spawn(['sh', '-c', 'pkcheck --action-id org.fedoraproject.FirewallD1.all
             firewall.readonly = false;
             firewall.debouncedEvent('changed');
             firewall.debouncedGetZones();
+        })
+        .catch(error => {
+            console.warn("pkcheck failed", error);
+
+            // Fall back to cockpit.permissions, "pkcheck" might not be available
+            const permission = cockpit.permission({ admin: true });
+            const update_permissions = () => {
+                console.log("permissions.allowed: ", permission.allowed);
+
+                if (permission.allowed) {
+                    firewall.readonly = false;
+                    firewall.debouncedEvent('changed');
+                    firewall.debouncedGetZones();
+                }
```

Comment:
The events should also be sent if the permission is not allowed, i.e. when dropping admin privileges again.  I. e. set `firewall.readonly = !permission.allowed`, and drop the `if`?